src/pages/smart-contracts/stacks-api-client-tutorial.md

```diff
@@ -0,0 +1,727 @@
+---
+title: Building a public registry
+description: Learn the basics of reading the state of the blockchain through the Stack Blockchain API
+duration: 60 minutes
+experience: intermediate
+tags:
+  - tutorial
+---
+
+## Introduction
+
+The [Stacks Blockchain API](/references/stacks-blockchain-api) is an API that helps app developers to view und use the state of the Stacks 2.0 Blockchain.
+
+In this tutorial you will extend a to-do app to use a public to-do list network on the blockchain. The to-do app was built in the [to-do app tutorial](/authentication/building-todo-app). You should have followed the instructions and the code ready on your local machine. It is also helpful if you have a basic understanding of Clarity as explained in the [counter tutorial](/smart-contracts/counter-tutorial).
+
+The network of public to-do lists is implemented by a clarity smart contract named `todo-registry`. The main contract function is `register` that allows to link a url of the public to-do list with the username. This information is stored in a contract data map and apps can use it for example to lookup urls by usernames.
+
+By the end of this tutorial, you will..
+
+- Have observed the progress of transaction processing
+- Have consumed APIs to show recent transactions, data map entries and read-only functions
+- Have experienced serializing and deserializing Clarity values
+
+## Prerequisits
+
+### Check the Stacks 2.0 status
+
+The Stacks 2.0 blockchain is currently in development and could experience resets and downtimes. To make sure you're not
+running into any challenges related to the status of the network, please open up the [Status Checker](http://status.test-blockstack.com/)
+and confirm that all systems are operational. If some systems seem to have issues, it is best to wait until they are back up before you proceed with the next steps.
+
+Furthermore, the to-do app will interact with a smart contract deployed as `ST25GX7MJA4FGEYYH5E7EBW30BYA4VG4VC1XETXXZ.todo-registry`, please verify that this contract exists by searching for it in the sandbox of the [testnet explorer](https://testnet-explorer.blockstack.org/sandbox?tab=contract-call).
+
+### Tutorials
+
+[@page-reference | grid-small]
+| /authentication/building-todo-app, /smart-contracts/counter-tutorial
+
+### Check your to-do app
+
+In your code repository of the to-do app launch the app by running
+
+```
+npm run start
+```
+
+## Registering a public url
+
+The Connect library (that is already used for authentication in the to-do app) provides also methods to create and send transactions on the blockchain. It uses the user's private key to sign transactions (not the user's app private key).
```

Comment:
ok.

I removed the private key all together. Somewhere we should have a guide or so that explains the difference between this tutorial and the signing transactions tutorial.

Reply:
what were you thinking of? can we include this callout somewhere?

Reply:
I was wrong here. The signing transactions tutorial is with Connect, like this tutorial. In my memory it was for wallet builders using private keys directly. 

@agraebe We could cut out the ## Registering a public URL section and refer to the signing transactions tutorial.

Reply:
Ah, got that! I think the best flow would be to show the code that is required and callout that transaction signing details are available in the other tutorial. So devs can follow along and get the full scoop by reading through the other page.

I'd also add the tutorial to the req chapter and call out that familiarity with contract signing is beneficial